tests/unit/projection-test.js

```diff
@@ -1296,1828 +1404,1830 @@ module('unit/projection', function(hooks) {
-          author: {
-            name: BOOK_AUTHOR_NAME_1,
-          },
-        });
+          run(() => {
+            set(baseModel, 'title', BOOK_TITLE_2);
+            set(baseModel, 'chapter-1', BOOK_CHAPTER_2);
+          });
 
-        // reify the nested model
-        get(record, 'author.name');
+          assert.equal(
+            get(projectedExcerpt, '_internalModel.currentState.isDirty'),
+            true,
+            'Expected projection to be made dirty'
+          );
+          assert.equal(
+            get(baseModel, '_internalModel.currentState.isDirty'),
+            true,
+            'Expected base model to be made dirty'
+          );
 
-        record.save();
-        return record;
-      });
+          run(() => {
+            projectedExcerpt.save();
+          });
 
-      // instead of involving adapter, just push the data and check things were correctly updated
-      run(() => {
-        this.store.push({
-          data: {
-            id: BOOK_ID,
-            type: BOOK_EXCERPT_PROJECTION_CLASS_PATH,
-            attributes: {
-              title: BOOK_TITLE_2,
-              author: {
-                name: BOOK_AUTHOR_NAME_2,
-              },
-            },
-          },
+          assert.equal(updateRecordCalls, 1, 'Expected one updateRecord call to be made');
+          assert.equal(
+            get(projectedExcerpt, '_internalModel.currentState.isDirty'),
+            false,
+            'The projection should have been saved'
+          );
+          assert.equal(
+            get(baseModel, '_internalModel.currentState.isDirty'),
+            true,
+            'The base model should still be dirty'
+          );
         });
       });
 
-      assert.equal(
-        get(projectedPreview, 'title'),
-        BOOK_TITLE_2,
-        'Expected preview projection to have received updated title'
-      );
-      assert.equal(
-        get(projectedPreview, 'author.name'),
-        BOOK_AUTHOR_NAME_2,
-        'Expected preview projection to have received updated author.name'
-      );
-    });
+      skip(`eachAttribute returns only white-listed properties`, function() {});
+      skip(`Creating a projection with an unloaded schema`, function() {});
+      skip(`Finding a projection with an unloaded schema`, function() {});
+      skip(`fetched schemas must be complete (projected types must also be included)`, function() {});
+    }
+  );
 
-    skip('we cannot create a new projection when existing recordData exists', function(assert) {
-      // pre-populate the store with a different projection and base-data for the ID we will attempt to create.
-      run(() => {
-        this.store.push({
-          data: {
-            id: BOOK_ID,
-            type: BOOK_EXCERPT_PROJECTION_CLASS_PATH,
-            attributes: {
-              title: BOOK_TITLE_2,
-            },
-          },
-        });
-      });
+  module('creating/updating projections with resolutions', function(hooks) {
```

Comment:
lets split this module out and make sure we test the old hooks (removed accidentally)

Reply:
done